Sources/OpenAPIRuntime/Conversion/Converter+Server.swift

```diff
@@ -281,17 +285,32 @@ public extension Converter {
     /// - Parameters:
     ///   - type: Type used to decode the data.
     ///   - data: Encoded body data.
+    ///   - strategy: A hint about which coding strategy to use.
     ///   - transform: Closure for transforming the Decodable type into a final type.
     /// - Returns: Deserialized body value.
     func bodyGetRequired<T: Decodable, C>(
         _ type: T.Type,
         from data: Data?,
+        strategy: CodingStrategy,
         transforming transform: (T) -> C
     ) throws -> C {
         guard let data else {
             throw RuntimeError.missingRequiredRequestBody
         }
-        let decoded = try decoder.decode(type, from: data)
+        let decoded: T
+        if let myType = T.self as? _StringParameterConvertible.Type,
+            strategy == .string
+        {
+            guard
+                let stringValue = String(data: data, encoding: .utf8),
+                let decodedValue = myType.init(stringValue)
+            else {
+                throw RuntimeError.failedToDecodeBody(type: T.self)
+            }
+            decoded = decodedValue as! T
+        } else {
+            decoded = try decoder.decode(type, from: data)
+        }
```

Comment:
Should we consider changing the semantics here to be stricter?

Specifically, if `bodyGetRequired(...)` is called with `strategy: .string` then I think it should attempt to use that strategy and fail if that wasn't the right choice. 

Right now it looks like if we ask for `.string` but the type is not `_StringParameterConvertible` we fallback to using Codable.

Maybe what we want is something like

```swift
        switch strategy {
        case .string:
            guard let stringConvertibleType = T.self as? _StringParameterConvertible.Type else {
                preconditionFailure("...")
            }
            guard
                let stringValue = String(data: data, encoding: .utf8),
                let decodedValue = myType.init(stringValue)
            else {
                throw RuntimeError.failedToDecodeBody(type: T.self)
            }
            decoded = decodedValue as! T
        case .json:
            decoded = try decoder.decode(type, from: data)
        }
```

Reply:
Possibly, although a few things to consider:
- the behavior had to be backwards compatible, so if you pass `.deferredToType`, you should get the "old" behavior
- the switch statement above doesn't handle `.deferredToType` - when you add it, you might see that you then have to duplicate the logic to decode both using string and codable methods inside that branch as well

Reply:
Hm, I think we can keep the backwards compatible functionality in the preserved functions that you've moved to `Deprecated.swift` and have this new function have the stricter semantics, especially for request and response body encoding because the content-type is explicitly mentioned in the OpenAPI document for each body.

Reply:
Did we address this in the latest commit?